tests/UnitTests/Rsp.QuestionSetService.UnitTests/Services/QuestionServiceTests/GetQuestionByVersionTests.cs

```diff
@@ -0,0 +1,111 @@
+    {
+        // Arrange
+        var questions = new List<Question>()
+        {
+            new() { QuestionId = "1", QuestionText = "Question 1", VersionId = "v1" },
+            new() { QuestionId = "2", QuestionText = "Question 2", VersionId = "v1" },
+            new() { QuestionId = "3", QuestionText = "Question 3", VersionId = "v2" }
+        };
+
+        Mocker
+            .GetMock<IQuestionRepository>()
+            .Setup(repo => repo.GetQuestions(It.IsAny<QuestionSpecification>()))
+            .ReturnsAsync(questions.Where(q => q.VersionId == "v1"));
+
+        // Act
+        var result = await Sut.GetQuestionsByVersion("v1");
+
+        // Assert
+        result.ShouldNotBeNull();
+        result.Count().ShouldBe(questions.Where(q => q.VersionId == "v1").Count());
+
+        // Verify
+        Mocker
+            .GetMock<IQuestionRepository>()
+            .Verify(repo => repo.GetQuestions(It.IsAny<QuestionSpecification>()), Times.Once);
+    }
+
+    [Fact]
+    public async Task GetQuestionsByVersion_ShouldReturnEmptyList_WhenNoQuestionsExist_ForTheGivenVersion()
+    {
+        // Arrange
+        Mocker
+            .GetMock<IQuestionRepository>()
+            .Setup(repo => repo.GetQuestions(It.IsAny<QuestionSpecification>()))
+            .ReturnsAsync([]);
+
+        // Act
+        var result = await Sut.GetQuestionsByVersion("v1");
+
+        // Assert
+        result.ShouldNotBeNull();
+        result.ShouldBeEmpty();
+
+        // Verify
+        Mocker
+            .GetMock<IQuestionRepository>()
+            .Verify(repo => repo.GetQuestions(It.IsAny<QuestionSpecification>()), Times.Once);
+    }
+
+    [Fact]
+    public async Task GetQuestionsByVersion_WithCategoryAndSection_ShouldReturnQuestions_ForTheGivenVersionCategorySection()
+    {
+        // Arrange
+        var questions = new List<Question>()
+        {
+            new() { QuestionId = "1", QuestionText = "Question 1", QuestionCategoryId = "Category 1", QuestionSectionId = "Section 1", VersionId = "v1" },
+            new() { QuestionId = "2", QuestionText = "Question 2", QuestionCategoryId = "Category 2", QuestionSectionId = "Section 2", VersionId = "v2" },
+            new() { QuestionId = "3", QuestionText = "Question 3", QuestionCategoryId = "Category 3", QuestionSectionId = "Section 3", VersionId = "v3" }
+        };
+
+        Mocker
+            .GetMock<IQuestionRepository>()
+            .Setup(repo => repo.GetQuestions(It.IsAny<QuestionSpecification>()))
+            .ReturnsAsync(questions.Where(q => q.VersionId == "v1" && q.QuestionSectionId == "Section 1" && q.QuestionCategoryId == "Category 1"));
+
+        // Act
+        var result = await Sut.GetQuestions();
+
+        // Assert
+        result.ShouldNotBeNull();
+        result.Count().ShouldBe(questions.Where(q => q.VersionId == "v1" && q.QuestionSectionId == "Section 1" && q.QuestionCategoryId == "Category 1").Count());
```

Comment:
Where is not needed here, you can just use Count. Also, when it's long line of code try moving down and chain the methods
```suggestion
        result
            .ShouldNotBeNull()
            .Count()
            .ShouldBe(questions.Count
            (
                q => q.VersionId == "v1" &&
                     q.QuestionSectionId == "Section 1" &&
                     q.QuestionCategoryId == "Category 1"
            ));
```